src/segmentedfilesystem/DataPacket.java

```diff
@@ -0,0 +1,31 @@
+package segmentedfilesystem;
+import java.util.Arrays;
+import java.net.DatagramPacket;
+
+public class DataPacket {
+    String stat = "data"; 
```

Comment:
What was this supposed to be for? There's a lot of "randomness" in this code that suggests that you really needed to step back and do some "big picture" work.